### ISSUE TITLE
robyn_clusters return error

### ISSUE THREAD

**Issue by andrewrjames** [facebookexperimental/Robyn#824]
## Describe issue
My model returns k=28 clusters and I want to use robyn_clusters to reduce the number of clusters to evaluate the model. 

## Provide reproducible example
cls <- robyn_clusters(
  input = OutputCollect,
  all_media = InputCollect$all_media,
  k = 5, limit = 5,
  # weights = c(1, 1, 1.5)
)

Result returns error 
<img width="826" alt="image" src="https://github.com/facebookexperimental/Robyn/assets/62786720/e5d5b8ed-84c9-4a76-b45b-2e0c5e6264dc">

## Environment & Robyn version
Make sure you're using the latest Robyn version before you post an issue.
- Check and share Robyn version: 3.10.4.9008
- R version 4.3.1

**Comment by gufengzhou** [comment]
please try directly specify k in the output func as `robyn_outputs(k = 11, ...)`

**Comment by andrewrjames** [comment]
@gufengzhou Ic. Actually I've increased iterations to 3500, used weibull_pdf, and added add_penalty_factor = True, but my decomp.rssd did not converge. I have added additional variables to explain the dependent variables, such as market demand, coupon as context_vars, as well as, push notification click as organic_vars too, but the decomp.rssd still did not converge.
1.  Can we conclude the model is sufficient enough since the median of the last 200 iterations did not change much?
<img width="788" alt="image" src="https://github.com/facebookexperimental/Robyn/assets/62786720/16ea9296-2e05-40aa-8079-251d30a31705">

2. Also when I looked at one of the results, the model seems to be overfit, adjusted r2 looks good for the training data, but it becomes lower for the test data. Is this something we should improve upon? 
<img width="491" alt="image" src="https://github.com/facebookexperimental/Robyn/assets/62786720/4eefb323-57cc-4b75-8125-277deb676ccd">


3. Additionally, the immediate vs carryover response showed a quite extreme result. For example, google sem has 100% immediate effect, while google display retargeting has complete 100% carry over effect. Intuitively, it does not really make sense to have an extreme result. Is this related to the weibull_pdf since it considers lagged effects?
<img width="491" alt="image" src="https://github.com/facebookexperimental/Robyn/assets/62786720/4eefb323-57cc-4b75-8125-277deb676ccd">

**Comment by gufengzhou** [comment]
1. well the "convergence" should be taken with a pinch of salt, but it's true that RSSD didnt move a lot. Maybe try to further increase iterations to 5k? probably worth trying. you can reduce bit trials for less duration.
2. Robyn doesn't forecast so far, so theoretically it doesn't matter much, and I know many just don't use ts_validation. In general, there could be many reasons why validation or test set don't fit well. Observe the actual vs predicted plot as a starter to see if the training set contains enough pattern to predict val/test at all.
3. For weibull adstock, it needs bit more finetuning of hyperparameters to fit your intuition. look at google display's peaks/lags, does it make sense that these channels have 7-8 weeks of lag? If not, what's your expectation/ hypothesis? Then set the hyperparameters for these channels to the proper range. Explaination of hyppar setting is in demo.R

**Comment by gufengzhou** [comment]
Please reopen if necessary.
